Scripts/RemediationScripts/Remediate-PTAControls.ps1

```diff
@@ -0,0 +1,915 @@
+<###
+# Overview:
+    This script is used to create PIM role assignment(s) with SC-ALT account for critical permanent role assignment(s) and PIM non SCALT role assignment(s) remove critical permanent role assignment(s) for which the PIM role assignment(s) is successfully created in the Subscription or in resource group.
+
+# Control ID:
+    Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access , Azure_Subscription_AuthZ_Dont_Grant_Persistent_Access_RG
+
+# Display Name:
+Do not grant permanent access for critical subscription level roles , Do not grant permanent access for critical resource group level roles
+
+# Prerequisites:
+    Owner and higher privileged role assignment on the Subscription is required and atleast one service adminstrator role assignment must be present on the subscription level.
+
+# Steps performed by the script:
+    To remediate:
+        1. Validate and install the modules required to run the script.
+        2. Get the list of critical permanent role assignment(s) in a Subscription or in a resource group.
+        3. Back up details of critical permanent role assignment(s) that are to be remediated.
+        4. create PIM role assignment(s) with SC-ALT account for critical permanent role assignment(s) and PIM non SCALT role assignment(s) remove critical permanent role assignment(s) for which the PIM role assignment(s) is successfully created in the Subscription or in resource group.
+   
+    NOTE: Please provide the output of dryrun only as input for remediation.
+
+
+# Instructions to execute the script:
+    To remediate:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to create PIM role assignment(s) with SC-ALT account for critical permanent role assignment(s) and remove critical permanent role assignment(s) for which the PIM role assignment(s) is successfully created in the Subscription or at resource group level. Refer `Examples`, below.
+
+# Examples:
+    To remediate:
+        1. To review the critical Permanent role assignment(s) on a Subscription that will be remediated:
+    
+           Create-PIMForPermanentRoleAssignments -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
```

Comment:
Just a thought - Can this be renamed to `Migrate-PermanentToPrivilegedRoleAssignments` or `Convert-PermanentToPrivilegedRoleAssignments`.

Reply:
Migrate-PermanentToPrivilegedRoleAssignments